Allow redacting on subdirs

naglepuff opened the pull request:
Fix #170 

Adds a new flag (`-r` or `--recursive`).

If this flag is provided to the `imagedephi run` command, then redaction will be performed on sub directories of the input directory. Additionally, the directory structure will be duplicated in the output directory.

manthey commented:
When you have --skip-rename turned on, it creates the directory structure (at least one deep) from the source, but the redacted files are not in those subdirectories.

naglepuff commented:
> When you have --skip-rename turned on, it creates the directory structure (at least one deep) from the source, but the redacted files are not in those subdirectories.

I forgot to update the parent directory of the output file when `--skip-rename` is used. I fixed that with 71119f8. 

I also updated the test to make sure this interaction doesn't break in the future with 231f38c.
